File: crates/gen-guest-rust/tests/records.rs
#[allow(unused_imports, unused_variables, dead_code)]
#[rustfmt::skip]
pub mod records {
    use ::tauri_bindgen_guest_rust::serde;
    use ::tauri_bindgen_guest_rust::bitflags;
    #[derive(serde::Serialize, serde::Deserialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub struct Empty {}
    /**A record containing two scalar fields
that both have the same type*/
    #[derive(serde::Serialize, serde::Deserialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub struct Scalars {
        ///The first field, named a
        pub a: u32,
        ///The second field, named b
        pub b: u32,
    }
    /**A record that is really just flags
All of the fields are bool*/
    #[derive(serde::Serialize, serde::Deserialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub struct ReallyFlags {
        pub a: bool,
        pub b: bool,
        pub c: bool,
        pub d: bool,
        pub e: bool,
        pub f: bool,
        pub g: bool,
        pub h: bool,
        pub i: bool,
    }
    #[derive(serde::Serialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub struct AggregatesParam<'a> {
        pub a: Scalars,
        pub b: u32,
        pub c: Empty,
        pub d: &'a str,
        pub e: ReallyFlags,
    }
    #[derive(serde::Deserialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub struct AggregatesResult {
        pub a: Scalars,
        pub b: u32,
        pub c: Empty,
        pub d: String,
        pub e: ReallyFlags,
    }
    pub type IntTypedef = i32;
    pub type TupleTypedef2 = (IntTypedef,);
    pub async fn tuple_arg(x: (char, u32)) {
        ::tauri_bindgen_guest_rust::invoke("records", "tuple_arg", &(x)).await.unwrap()
    }
    pub async fn tuple_result() -> (char, u32) {
        ::tauri_bindgen_guest_rust::invoke("records", "tuple_result", &()).await.unwrap()
    }
    pub async fn empty_arg(x: Empty) {
        ::tauri_bindgen_guest_rust::invoke("records", "empty_arg", &(x)).await.unwrap()
    }
    pub async fn empty_result() -> Empty {
        ::tauri_bindgen_guest_rust::invoke("records", "empty_result", &()).await.unwrap()
    }
    pub async fn scalar_arg(x: Scalars) {
        ::tauri_bindgen_guest_rust::invoke("records", "scalar_arg", &(x)).await.unwrap()
    }
    pub async fn scalar_result() -> Scalars {
        ::tauri_bindgen_guest_rust::invoke("records", "scalar_result", &())
            .await
            .unwrap()
    }
    pub async fn flags_arg(x: ReallyFlags) {
        ::tauri_bindgen_guest_rust::invoke("records", "flags_arg", &(x)).await.unwrap()
    }
    pub async fn flags_result() -> ReallyFlags {
        ::tauri_bindgen_guest_rust::invoke("records", "flags_result", &()).await.unwrap()
    }
    pub async fn aggregate_arg(x: AggregatesParam<'_>) {
        ::tauri_bindgen_guest_rust::invoke("records", "aggregate_arg", &(x))
            .await
            .unwrap()
    }
    pub async fn aggregate_result() -> AggregatesResult {
        ::tauri_bindgen_guest_rust::invoke("records", "aggregate_result", &())
            .await
            .unwrap()
    }
    pub async fn typedef_inout(e: TupleTypedef2) -> i32 {
        ::tauri_bindgen_guest_rust::invoke("records", "typedef_inout", &(e))
            .await
            .unwrap()
    }
}

File: crates/gen-guest-dart/tests/records.ts


export interface Empty {  }
/**
 * A record containing two scalar fields 
 * that both have the same type 
*/
export interface Scalars { /**
 * The first field, named a 
*/
a: number,
/**
 * The second field, named b 
*/
b: number,
 }
/**
 * A record that is really just flags 
 * All of the fields are bool 
*/
export interface ReallyFlags { 
a: boolean,

b: boolean,

c: boolean,

d: boolean,

e: boolean,

f: boolean,

g: boolean,

h: boolean,

i: boolean,
 }

export interface Aggregates { 
a: Scalars,

b: number,

c: Empty,

d: string,

e: ReallyFlags,
 }

export type IntTypedef = number;

export type TupleTypedef2 = [IntTypedef];


            
            export async function tupleArg (x: [string, number])  {
            }
        
            
            export async function tupleResult () : Promise<[string, number]> {
            }
        
            
            export async function emptyArg (x: Empty)  {
            }
        
            
            export async function emptyResult () : Promise<Empty> {
            }
        
            
            export async function scalarArg (x: Scalars)  {
            }
        
            
            export async function scalarResult () : Promise<Scalars> {
            }
        
            
            export async function flagsArg (x: ReallyFlags)  {
            }
        
            
            export async function flagsResult () : Promise<ReallyFlags> {
            }
        
            
            export async function aggregateArg (x: Aggregates)  {
            }
        
            
            export async function aggregateResult () : Promise<Aggregates> {
            }
        
            
            export async function typedefInout (e: TupleTypedef2) : Promise<number> {
            }
        